src/hcupc_spmd.cpp

```diff
@@ -137,13 +141,7 @@ int barrier() {
  * This is adapted from:
  * Olivier, S., Prins, J.: Scalable dynamic load balancing using upc. In: ICPP. pp.123�131 (2008)
  */
-upcxx::shared_var<int> cb_cancel;
-upcxx::shared_var<int> cb_count;
-upcxx::shared_var<int> cb_done;
-upcxx::shared_lock cb_lock;
-bool workAvailInit = false;
-#define CB_LOCK		cb_lock.lock()
-#define CB_UNLOCK	cb_lock.unlock()
+static upcxx::shared_array<upcxx::atomic<unsigned int > > cb_count;
```

Comment:
This is good, but need to check if works fine with large node counts. I expect it should